File: truevfs-ext/truevfs-ext-insight/src/main/scala/net/java/truevfs/ext/insight/I5tIoStatistics.scala
/*
 * Copyright (C) 2005-2015 Schlichtherle IT Services.
 * All rights reserved. Use is subject to license terms.
 */
package net.java.truevfs.ext.insight

import javax.annotation.concurrent.ThreadSafe

/**
 * A controller for [[net.java.truevfs.ext.insight.stats.IoStatistics]].
 *
 * @author Christian Schlichtherle
 */
@ThreadSafe
private final class I5tIoStatistics(mediator: I5tMediator, offset: Int)
extends I5tStatistics(mediator, offset) {

  def newView: I5tStatisticsView = new I5tIoStatisticsView(this)
}

File: truevfs-ext/truevfs-ext-insight/src/main/java/net/java/truevfs/ext/insight/I5tIoStatistics.java
/*
 * Copyright (C) 2005-2012 Schlichtherle IT Services.
 * All rights reserved. Use is subject to license terms.
 */
package net.java.truevfs.ext.insight;

import javax.annotation.concurrent.ThreadSafe;
import net.java.truevfs.ext.insight.stats.IoStatistics;

/**
 * A controller for {@linkplain IoStatistics I/O statistics}.
 * 
 * @see    JmxSyncStatistics
 * @author Christian Schlichtherle
 */
@ThreadSafe
final class I5tIoStatistics extends I5tStatistics {

    I5tIoStatistics(I5tMediator mediator, int offset) {
        super(mediator, offset);
    }

    @Override
    I5tStatisticsView newView() { return new I5tIoStatisticsView(this); }
}
